server/src/core/models/embeds/inventory.embed.ts

```diff
@@ -0,0 +1,6 @@
+import { FK } from '../../../shared/common/persistance/db.types';
+import { Int } from '../../../shared/types/base.types';
+import Item from '../item.model';
+
+type Inventory = Array<{ item: FK<Item>; stacks: Int }>;
```

Comment:
I think you might also need a position, if we want a grid-like inventory or to keep any kind of order

Reply:
we do have the index of the array